projects/ContextRouting/context.yaml

```diff
@@ -0,0 +1,144 @@
+
+AWSTemplateFormatVersion: "2010-09-09"
+Description: Sample context tracking with Amazon Connect
+Parameters: 
+  TrackerDynamoTableName: 
+    Type: String
+    Default: ContextTrackingTable
+    Description: Provide a name for your context tracker DB
+  KinesisCTRStreamARN: 
+    Type: String
+    Default: REPLACEME
+    Description: Provide the ARN for the Kinesis stream that receives your contact trace records
+  TrackerRecordTimeToLive: 
+    Type: Number
+    Default: 300
+    Description: Amount of time, in seconds, before the tracker expires. Default is 5 minutes (300 seconds)
+  AWSRegion: 
+    Type: String
+    Default: us-east-1
+    AllowedValues:
+      - us-east-1
+      - us-west-2
+      - ap-southeast-1
+      - ap-southeast-2
+      - ap-northeast-1
+      - eu-central-1
+      - eu-west-2
+    Description: The region code that you are deploying to. Must be a region that supports Amazon Connect
+Resources:
+  ContextSampleRole:
+    Type: 'AWS::IAM::Role'
+    Properties:
+      Description: IAM Role that provides the access required for this sample
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service:
+              - lambda.amazonaws.com
+            Action:
+              - 'sts:AssumeRole'
+      Path: /
+      ManagedPolicyArns: 
+        - arn:aws:iam::aws:policy/AmazonKinesisReadOnlyAccess
+        - arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole
+      Policies:
+        - PolicyName: ContextLambdaAddOn
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
+                  - dynamodb:PutItem
+                  - dynamodb:GetItem
+                  - dynamodb:DeleteItem
+                  - dynamodb:UpdateItem
+                  - dynamodb:GetRecords
+                Resource: '*'
+  ContextTrackerTable:
+    Type: AWS::DynamoDB::Table
+    Properties:
+      AttributeDefinitions:
+        - 
+          AttributeName: "caller_id"
+          AttributeType: "S"
+      KeySchema:
+        - 
+          AttributeName: "caller_id"
+          KeyType: "HASH"
+      ProvisionedThroughput:
```

Comment:
should this use the pay per request billing mode?

Reply:
In this case, keeping it provisioned allows me to keep it under free tier. Switching to on-demand means that it could spike up and exceed free tier.

Reply:
gotcha, it's just that while this is true in isolation - when every single snippet and template uses the same 5 reqs/second default customers end up spending money on provisioned throughput that they never really need to.

You might want to add a note in the readme about pay per request option on the DB if the call center has different active hours